libs/redisconsumer/consumer.go

```diff
@@ -114,21 +114,42 @@ func (redisClient *RedisClient) CreateStream(ctx context.Context, stream, consum
 }
 
 // AddMessage to the specified redis stream
-func (redisClient *RedisClient) AddMessages(ctx context.Context, stream string, message ...interface{}) error {
+func (redisClient *RedisClient) AddMessages(ctx context.Context, stream string, messages ...interface{}) error {
 	pipe := ((*redis.Client)(redisClient)).Pipeline()
-
-	for _, v := range message {
-		pipe.XAdd(
-			ctx, &redis.XAddArgs{
-				Stream: stream,
-				Values: map[string]interface{}{
-					"data": v}},
-		)
+	var err error
+
+	// to avoid errors enqueuing large message sets, enqueue them in chunks
+	for _, messages := range chunkMessages(messages) {
+		// loop again so that each message gets its own record
+		for _, message := range messages {
+			pipe.XAdd(
+				ctx, &redis.XAddArgs{
+					Stream: stream,
+					Values: map[string]interface{}{
+						"data": message}},
+			)
+		}
+		_, err = pipe.Exec(ctx)
+		if err != nil {
+			return err
+		}
 	}
-	_, err := pipe.Exec(ctx)
 	return err
 }
 
+func chunkMessages(messages []interface{}) [][]interface{} {
+	var chunks [][]interface{}
+	chunkSize := 10000
+	for i := 0; i < len(messages); i += chunkSize {
+		end := i + chunkSize
+		if len(messages) < end {
+			end = len(messages)
+		}
+		chunks = append(chunks, messages[i:end])
+	}
+	return chunks
+}
```

Comment:
This can be slightly tidied up and micro-, if you will, optimised. The `messages` slice is passed in, so it's likely that calling `len` multiple times on it will [incur overhead](https://stackoverflow.com/questions/26634554/go-multiple-len-calls-vs-performance).

Additionally, the size of a portion is better passed in as a parameter, so the code is easier to adjust at the call site.

1. If it were me, I would have written it as follows, also improving on readability:

```suggestion
func chunkMessages(lim int, messages []interface{}) [][]interface{} {
	total := len(messages)
	if total == 0 {
		return nil
	}

	result := make([][]interface{}, 0, 1)
	for start, stop := 0, 0; start < total; start = stop {
		stop += lim
		if stop > total {
			stop = total
		}

		result = append(result, messages[start:stop])
	}

	return result
}
```

2. Or even in generic form:

```suggestion
func splitSlice[T any](lim int, slice []T) [][]T {
	total := len(slice)
	if total == 0 {
		return nil
	}

	result := make([][]T, 0, 1)
	for start, stop := 0, 0; start < total; start = stop {
		stop += lim
		if stop > total {
			stop = total
		}

		result = append(result, slice[start:stop])
	}

	return result
}
```

3. Your current version could be slightly changed like this:

```suggestion
func chunkMessages(messages []interface{}) [][]interface{} {
	const chunkSize = 10000

	total := len(messages)
	var chunks [][]interface{}

	for i := 0; i < total; i += chunkSize {
		end := i + chunkSize
		if total < end {
			end = total
		}

		chunks = append(chunks, messages[i:end])
	}

	return chunks
}
```

Reply:
Good recommendation